File: src-tauri/src/settings/dto/update_settings_request.rs
use std::path::PathBuf;

use serde::{Deserialize, Serialize};

use crate::settings::value_objects::{settings_profile::SettingsProfile, theme::Theme};

#[derive(Default, Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateSettingsRequest {
    pub base_database_directory: Option<PathBuf>,
    pub profile: Option<SettingsProfile>,

    pub theme: Option<Theme>,
    pub zoom_percentage: Option<f64>,
    pub auto_sync: Option<bool>,

    pub enable_ai: Option<bool>,
    pub ollama_model_name: Option<Option<String>>,
    pub ollama_embeddings_model_name: Option<Option<String>>,
}

File: src/api/settings/models/updateSettingsRequest.ts
import { Theme } from "../dto/updateSettingsRequestDto";

export default interface UpdateSettingsRequest {
	baseDatabaseDirectory: string | null;

	theme: Theme | null;
	zoomPercentage: number | null;
	autoSync: boolean | null;

	enableAi: boolean | null;
	ollamaModelName: string | null;
	ollamaEmbeddingsModelName: string | null;
}
